lib/charms/parca_k8s/v0/parca_scrape.py

```diff
@@ -437,7 +374,10 @@ def jobs(self) -> list:
         """
         scrape_jobs = []
 
-        for relation in self._charm.model.relations[self._relation_name]:
+        _validate_relation_by_interface_and_direction(charm, self._relation_name, RELATION_INTERFACE_NAME,
```

Comment:
off topic from the main idea of this PR... but does this check add value?  iiuc this protects against:
* writing metadata.yaml with:
  * `interface-xyz` when it should be `interface-123`
  * `relation-name-xyz` when it should be `relation-name-123`
  * the correct direction (provider is listed as a provider, etc)

Aren't all of those static after compile time, and any misconfigurations to them should break unit tests if coverage is satisfactory?  

I don't see why this should happen every time the charm works